src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp

```diff
@@ -387,18 +398,49 @@ class boss_archimonde : public CreatureScript
                             DoAction(ACTION_ENRAGE);
                     events.ScheduleEvent(EVENT_DISTANCE_CHECK, 5s);
                     break;
+                case EVENT_PROTECTION_OF_ELUNE: // hp below 10% only cast finger of death
+                    events.Reset();
+                    events.ScheduleEvent(EVENT_HAND_OF_DEATH, 1s);
+                    events.ScheduleEvent(EVENT_FINGER_OF_DEATH_LAST_PHASE, 1s);
+                    events.ScheduleEvent(EVENT_SUMMON_WHISP, 1s);
+                    CastProtectionOfElune(true);
+                    break;
                 case EVENT_SUMMON_WHISP:
                     DoSpawnCreature(NPC_ANCIENT_WISP, float(rand32() % 40), float(rand32() % 40), 0, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15s);
                     ++WispCount;
                     if (WispCount >= 30)
+                    {
                         me->KillSelf();
+                        return;
+                    }
                     events.ScheduleEvent(EVENT_SUMMON_WHISP, 1500ms);
                     break;
+                case EVENT_FINGER_OF_DEATH_LAST_PHASE:
+                    DoCast(SelectTarget(SelectTargetMethod::Random, 0), SPELL_FINGER_OF_DEATH_LAST_PHASE);
+                    events.ScheduleEvent(EVENT_FINGER_OF_DEATH_LAST_PHASE, 1s);
+                    break;
                 default:
                     break;
             }
         }
 
+        void CastProtectionOfElune(bool apply)
```

Comment:
This whole method can be removed, just cast SPELL_PROTECTION_OF_ELUNE once, without AddAura